src/esp/gfx/CubeMap.cpp

```diff
@@ -0,0 +1,458 @@
+    switch (type) {
+      case TextureType::Color: {
+        Mn::Image2D image =
+            textures_[type]->image(convertFaceIndexToCubeMapCoordinate(iFace),
+                                   0, {getPixelFormat(type)});
+        filename = Cr::Utility::formatString("{}.{}.{}.png", imageFilePrefix,
+                                             getTextureTypeFilenameString(type),
+                                             coordStrings[iFace]);
+        if (!converter->exportToFile(image, filename)) {
+          return false;
+        }
+      } break;
+
+      case TextureType::Depth: {
+        filename = Cr::Utility::formatString("{}.{}.{}.hdr", imageFilePrefix,
+                                             getTextureTypeFilenameString(type),
+                                             coordStrings[iFace]);
+        Mn::Image2D image = textures_[type]->image(
+            convertFaceIndexToCubeMapCoordinate(iFace), 0,
+            {Mn::GL::PixelFormat::DepthComponent, Mn::GL::PixelType::Float});
+        Mn::ImageView2D depthAsRedChannel{
+            image.storage(), Mn::PixelFormat::R32F, image.size(), image.data()};
+        if (!converter->exportToFile(depthAsRedChannel, filename)) {
+          return false;
+        }
+        // XXX
+        // display 1st 4 rgbs
+        Mn::Debug{} << "1st 4 rgb ==== SAVE"
+                    << Cr::Containers::arrayCast<float>(
+                           image.data().prefix(4 * 3 * 4));
+      } break;
+    }
+    CORRADE_ASSERT(!filename.empty(),
+                   "CubeMap::saveTexture(): Unknown texture type.", false);
+
+    LOG(INFO) << "Saved image " << iFace << " to " << filename;
+  }  // for
+
+  return true;
+}
+#endif
+
+void CubeMap::loadTexture(TextureType type,
+                          const std::string& imageFilePrefix,
+                          const std::string& imageFileExtension) {
+  textureTypeSanityCheck(flags_, type, "CubeMap::loadTexture():");
+
+  // set the alias of the texture
+  Mn::GL::CubeMapTexture* texture = nullptr;
+  switch (type) {
+    case TextureType::Color:
+      texture = textures_[TextureType::Color].get();
+      break;
+
+    case TextureType::Depth:
+      texture = textures_[TextureType::Depth].get();
+      break;
+  }
+  CORRADE_ASSERT(texture, "CubeMap::loadTexture(): Unknown texture type.", );
+
+  // plugin manager used to instantiate importers which in turn are used
+  // to load image data
+  Cr::PluginManager::Manager<Mn::Trade::AbstractImporter> manager;
+  std::string importerName{"AnyImageImporter"};
+  if (type == TextureType::Depth) {
+    importerName = "StbImageImporter";
+  }
+  Cr::Containers::Pointer<Mn::Trade::AbstractImporter> importer =
+      manager.loadAndInstantiate(importerName);
+  CORRADE_INTERNAL_ASSERT(importer);
+  if (type == TextureType::Depth) {
+    // Override image channel count. Because depth info should be R32F
+    importer->configuration().setValue("forceChannelCount", 1);
```

Comment:
@mosra:

I just updated the PR using `forceChannelCount` based on my understanding. Still no luck here. 
The data saved in the `saveTexture` are different from the data loaded from the `loadTexture` (I printed out the first 12 floats).

Reply:
> The data saved in the saveTexture are different from the data loaded from the loadTexture

That's inherent to the HDR format, as I on Slack few weeks ago -- see https://en.wikipedia.org/wiki/RGBE_image_format for reference.

> Note that I did not touch the saveTexturefunction which was still using anyImageConverterto dump out depth info. Should I use stbImageConverter`?

Ah, yeah... if you still get three channels on input and not just one, you need to use `StbImageImporter` directly, not the `AnyImageImporter` proxy. (Sorry about that. I have a TODO there, the configuration values aren't propagated from `AnyImageImporter` currently because of some missing APIs in Corrade.)